### PR TITLE
fix: rpc creation no longer blocks the gateway

### LINKED ISSUE #106: bug: services block on starting ctl rpc server
Since the watch command, the gateway (and presumably other services) are unresponsive. On gateway it's due to https://github.com/n0-computer/iroh/blob/main/iroh-gateway/src/core.rs#L87, the call never returns as it's basically starting the tonic server on the current thread and essentially waiting for it to halt. Should probably spin it off similar to https://github.com/n0-computer/iroh/blob/main/iroh-gateway/src/main.rs#L49

### PR DESCRIPTION
**Pull request by ramfox**
Also, adds `/health` endpoint that returns both status code OK, and string "OK".

closes #106

### CODE REVIEW

File: iroh-gateway/src/core.rs [hunk 1]
```diff
@@ -551,3 +560,45 @@ async fn middleware_error_handler(
...
+        let handler = Core::new(config, gw_metrics).await.unwrap();
+        let core_task = tokio::spawn(async move {
+            handler.serve().await;
+        });
+
+        let res = hyper::Client::new()
+            .get(hyper::Uri::from_static("http://localhost:9001/health"))
+            .await
+            .unwrap();
+        assert_eq!(StatusCode::OK, res.status());
```
Comment (dignifiedquire) [inline iroh-gateway/src/core.rs:601]: this should also check for the content

File: iroh-gateway/src/core.rs [hunk 2]
```diff
@@ -551,3 +560,45 @@ async fn middleware_error_handler(
...
+        core_task.abort();
+    }
```
Comment (dignifiedquire) [inline iroh-gateway/src/core.rs:603]: should also call `core_task.await`, to wait for shutdown to complete

**Comment by Arqu** [comment]
Got nothing to add, we've had the ports discussion on discord & think the setup for the test is as good as it gets without making explicit additions to make this case easier.

**Comment by ramfox** [comment]
@Arqu want to bring your attention to one thing:
In order to get the address of the gateway server, we need to make serving a 2 step process. One step to build the server, and a separate step to run it (aka call `server.await`). Needed to add this so we can get the port # on random port bindings.
The change just mimics what we already have/the api that already exists in axum, it just separates the building from the listening.

**Review by dignifiedquire — APPROVED** [review]

**Pull request merged**